rotkehlchen/icons.py

```diff
@@ -160,7 +160,7 @@ def _assets_with_coingecko_id(self) -> Dict[str, str]:
         """Create a mapping of all the assets identifiers to their coingecko id if it is set"""
         querystr = """
         SELECT A.identifier, B.coingecko from assets as A JOIN common_asset_details as B
-        ON B.identifier = A.identifier A.type != ? AND B.coingecko IS NOT NULL AND B.coingecko != ""
+        ON B.identifier = A.identifier WHERE A.type != ? AND B.coingecko IS NOT NULL AND B.coingecko != ""
```

Comment:
Wait wait ... if there is such an error and we have not seen this at all so far this means this is (1) not used very often and (2) there is no test for this.

Or we would have seen this glaring error.

That's not how you fix stuff. Don't make an one word PR for something that is obviously broken. Create a test that shows why this is broken, reproduces the problem and detects it and then fix it and see test passes.

Reply:
Got you. I think we have not seen this since:
1. This method is used in a task that runs with `exception_is_error=False` https://github.com/rotki/rotki/blob/966c2b3eb8ec1e7712be0909d16c5dda1a1e3d65/rotkehlchen/rotkehlchen.py#L386
2. From what I found, it runs only at startup, so this exception doesn't happen really often
3. Most of the icons are cached, so perhaps we don't notice a difference

Reply:
Weird ... but now I know why I am missing many icons.

I guess we would also notice it if we pressed the refresh coingecko icon cache button? Or we don't have such a button? I know for sure we have per-icon refresh but perhaps we don't have a "refresh all button".

For the exception_is_error, you can still notice it but only if you look at the logs. Which is why logs are so important for us to analyze.

```
log.warning(f'{first_line} but that is not treated as an error')
```

Sometimes when I get a user's logs I consider it a small treasure. Seeing how they use rotki and all our funny logs inside it may (or may not :disappointed: ) uncover interesting behavior we had not thought before.

Reply:
Yes, the logs is how I found this problem!